### PR TITLE
Sign off

### PR DESCRIPTION
**Pull request by sam94dion**
Signed-off-by: Samuel <sam94dion@gmail.com>

### CODE REVIEW

**Comment by sam94dion** [comment]
Hi,

So I did a rebase to avoid signing of all my previous commits. Sorry If that is not the best way. Doing a rebase is what I found on stackoverflow.

Added functionalities :

1. If you want to test a subprogram but the sub program is passed a linkage section you would want to stub that part for the program to be able to run without dependences.

example:

```
 LINKAGE SECTION.  
 COPY COPY002.
 PROCEDURE DIVISION.
```


TO 

```
      * LINKAGE SECTION.
       
       COPY COPY002.

       PROCEDURE DIVISION.
```

2. You might declare SQL cursors within your working-section which will need to be stubbed

example:

```
           EXEC SQL  
              DECLARE NAME-CUR CURSOR FOR  
              SELECT FIRST_NAME, LAST_NAME FROM TEXEM
           END-EXEC.
```

TO

```
      *      EXEC SQL  
      *        DECLARE NAME-CUR CURSOR FOR  
      *        SELECT FIRST_NAME, LAST_NAME FROM TEXEM
      *      END-EXEC.
```
3. If you have SQL copybook you might want to use the variables within the copybook but discard other information.
example:

```
       EXEC SQL INCLUDE TEXEM  END-EXEC.
```

TO

```
       01  TEXEM.                                                               
           10 FIRST-NAME           PIC X(10).                                   
           10 LAST-NAME            PIC X(10).                                   
           10 TMS-CREA             PIC X(26).     
```

**Comment by sam94dion** [comment]
@dakaa16 

I didn't want to make a instant of reader in the generate class to use readAsOneStatement() that is why I created the getter()

**Review by dakaa16 — CHANGES_REQUESTED** [review]
Hi @sam94dion  - Happy new year! :D
There are some minor issues, that I would like you to fix. In addition to that, please do not touch the zip files, as they are the different distribution versions, that are meant for the public to download. These should not be deleted, and a new version should only be added, when a new release is planned.

I apologize for any inconvenience the late reply might cause, and I hope that you are still interested in helping us getting you changes into the next release of cobol check.

**Comment by sam94dion** [comment]
Hi @dakaa16 I'll get those fixed in a bit :)

File: .gitignore [hunk 1]
```diff
@@ -120,3 +120,8 @@ nb-configuration.xml
 ## OS X
 ##############################
 .DS_Store
+output/testResults.txt
+approval-test-actual.txt
+default.conf
```
[the remark this answers is not in the document]
Comment (sam94dion) [reply .gitignore:125]: This file is the default.conf that you need to run gnu Cobol. I could take it out and add it in my environment variables

File: src/main/java/org/openmainframeproject/cobolcheck/features/interpreter/CopybookExpander.java [hunk 2]
```diff
@@ -17,18 +20,25 @@
 /**
  * Expand copybooks referenced by the code under test.
  *
- * In the general use case, COPY statements are left alone and the compiler handles expansion.
+ * In the general use case, COPY statements are left alone and the compiler
```
[the remark this answers is not in the document]
Comment (sam94dion) [reply src/main/java/org/openmainframeproject/cobolcheck/features/interpreter/CopybookExpander.java:23]: fixed

File: cobolcheck.cmd [hunk 3]
```diff
@@ -1,2 +1,2 @@
 @echo off
-java -jar bin\cobol-check-@VERSION@.jar %*
\ No newline at end of file
+java -jar bin\cobol-check-0.2.4.jar %*
```
[the remark this answers is not in the document]
Comment (sam94dion) [reply cobolcheck.cmd]: fixed

File: .gitignore (hunk 1 above)
[the remark this answers is not in the document]
Comment (sam94dion) [reply .gitignore:125]: I left it there please tell me if this is a problem

**Comment by sam94dion** [comment]
Done should be better :)

**Comment by Rune-Christensen** [comment]
Hi @sam94dion 
These changes were implemented in pull request https://github.com/openmainframeproject/cobol-check/pull/278 
I am therefore closing this PR.
Thank you :-)

**Pull request closed without merging**